mlflow/server/handlers.py

```diff
@@ -16,29 +16,39 @@
     UpdateRun, LogMetric, LogParam, SetTag, ListExperiments, \
     DeleteExperiment, RestoreExperiment, RestoreRun, DeleteRun, UpdateExperiment, LogBatch
 from mlflow.store.artifact_repository_registry import get_artifact_repository
-from mlflow.tracking.utils import _is_database_uri, _is_local_uri
+from mlflow.store.dbmodels.db_types import DATABASE_ENGINES
+from mlflow.tracking.registry import TrackingStoreRegistry
 from mlflow.utils.proto_json_utils import message_to_json, parse_dict
 from mlflow.utils.search_utils import SearchFilter
 from mlflow.utils.validation import _validate_batch_log_api_req
 
 _store = None
 
 
+def _get_file_store(store_uri, artifact_uri):
+    from mlflow.store.file_store import FileStore
+    return FileStore(store_uri, artifact_uri)
+
+
+def _get_sqlalchemy_store(store_uri, artifact_uri):
+    from mlflow.store.sqlalchemy_store import SqlAlchemyStore
+    return SqlAlchemyStore(store_uri, artifact_uri)
+
+
+_tracking_store_registry = TrackingStoreRegistry()
+_tracking_store_registry.register('', _get_file_store)
+_tracking_store_registry.register('file', _get_file_store)
+for scheme in DATABASE_ENGINES:
```

Comment:
can these lines also be moved into `registry` package under a helper function. `tracking/utils.py` also uses registry in the same manner. So if we have a factory method like so, could be useful for all clients of this

```
def get_tracking_store_registry():
  tsr = TrackingStoreRegistry()
  tsr.register('', _get_file_store)
  tsr.register('file', _get_file_store)
  for scheme in DATABASE_ENGINES:
    tsr.register(scheme, _get_sqlachemy_store)
  ...
```